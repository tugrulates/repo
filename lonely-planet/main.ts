// deno-lint-ignore-file no-console
import { Command } from "@cliffy/command";
import { Table } from "@cliffy/table";
import { displayVersion } from "@roka/package/version";
import { LonelyPlanetClient } from "./client.ts";
import { EMOJIS } from "./data.ts";
import type { Attraction, Destination } from "./types.ts";

/** Breadcrumb text from a location. */
function breadcrumb(document: Destination | Attraction) {
  return `[ ${
    document.breadcrumb.map((breadcrumb) => breadcrumb.title).join(" > ")
  } ]`;
}

async function getCommand() {
  return new Command()
    .name("lonely-planet")
    .description("Explores data from Lonely Planet.")
    .version(await displayVersion())
    .example("lonely-planet big sur", "Search destinations for 'big sur'.")
    .example("lonely-planet --attractions amsterdam", "Search attractions.")
    .example("lonely-planet --stories amsterdam", "Search stories.")
    .example("lonely-planet --destinations --attractions --stories", "All.")
    .example("lonely-planet --json | jq", "Stream destinations as json.")
    .arguments("[keywords...:string]")
    .option("--destinations", "Include destinations in the results.")
    .option("--attractions", "Include attractions in the results.")
    .option("--stories", "Include stories in the results.")
    .option("--json", "Output the search results as concatenated JSON.")
    .action(
      async ({ destinations, attractions, stories, json }, ...keywords) => {
        if (!destinations && !attractions && !stories) destinations = true;
        const client = new LonelyPlanetClient();
        const rows: string[][] = [];
        if (destinations) {
          for await (const doc of client.searchDestinations(keywords)) {
            if (json) console.log(JSON.stringify(doc));
            else rows.push([EMOJIS[doc.type], doc.title, breadcrumb(doc)]);
          }
        }
        if (attractions) {
          for await (const doc of client.searchAttractions(keywords)) {
            if (json) console.log(JSON.stringify(doc));
            else rows.push([EMOJIS[doc.type], doc.title, breadcrumb(doc)]);
          }
        }
        if (stories) {
          for await (const doc of client.searchStories(keywords)) {
            if (json) console.log(JSON.stringify(doc));
            else rows.push([EMOJIS[doc.type], doc.title]);
          }
        }
        Table.from(rows).render();
      },
    );
}

/** CLI entrypoint. */
export async function main(args: string[]) {
  const command = await getCommand();
  await command.parse(args);
}
