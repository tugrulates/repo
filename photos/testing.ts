/**
 * This module provides test photos to test the `photos` package.
 *
 * @module testing
 */

import { pool } from "@roka/async/pool";
import { assertExists } from "@std/assert";
import { copy } from "@std/fs";
import { join } from "@std/path";
import { write, type WriteOptions } from "@tugrulates/photos/exif";
import { type Photo, photo } from "@tugrulates/photos/photo";

/**
 * Creates a temporary photo.
 *
 * @param options Update photo with given tags.
 * @returns A disposable photo with variants.
 */
export async function tempPhoto(
  options?: WriteOptions,
): Promise<Photo & AsyncDisposable> {
  const directory = await Deno.makeTempDir();
  assertExists(import.meta.dirname);
  await copy(
    join(import.meta.dirname, "__testdata__", "floating-around"),
    directory,
    { overwrite: true },
  );
  let result = await photo(directory);
  if (options) {
    await write(result.path, options);
    await pool(result.variants, (variant) => write(variant.path, options));
    result = await photo(directory);
  }
  return Object.assign(await photo(directory), {
    [Symbol.asyncDispose]: () => Deno.remove(directory, { recursive: true }),
  });
}
