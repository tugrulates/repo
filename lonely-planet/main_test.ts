import { mockFetch } from "@roka/http/testing";
import { fakeConsole } from "@roka/testing/fake";
import { assertSnapshot } from "@std/testing/snapshot";
import { main } from "./main.ts";

Deno.test("lonely-planet [keywords...]", async (t) => {
  using console = fakeConsole();
  using _fetch = mockFetch(t);
  await main(["the", "netherlands"]);
  await assertSnapshot(t, console.calls);
});

Deno.test("lonely-planet [keywords...] --json", async (t) => {
  using console = fakeConsole();
  using _fetch = mockFetch(t);
  await main(["the", "netherlands", "--json"]);
  await assertSnapshot(t, console.calls);
});

Deno.test("lonely-planet --destinations", async (t) => {
  using console = fakeConsole();
  using _fetch = mockFetch(t);
  await main(["--destinations", "amsterdam"]);
  await assertSnapshot(t, console.calls);
});

Deno.test("lonely-planet --attractions", async (t) => {
  using console = fakeConsole();
  using _fetch = mockFetch(t);
  await main(["--attractions", "haarlem"]);
  await assertSnapshot(t, console.calls);
});

Deno.test("lonely-planet --stories", async (t) => {
  using console = fakeConsole();
  using _fetch = mockFetch(t);
  await main(["--stories", "utrecht"]);
  await assertSnapshot(t, console.calls);
});
