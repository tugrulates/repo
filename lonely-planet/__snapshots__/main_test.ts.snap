export const snapshot = {};

snapshot[`lonely-planet [keywords...] 1`] = `
[
  {
    data: [
      "🏳️ The Netherlands                      [ Europe ]                              
🏞️ Southeastern Netherlands             [ Europe > The Netherlands ]            
🏙️ Amsterdam                            [ Europe > The Netherlands ]            
🏙️ Rotterdam                            [ Europe > The Netherlands ]            
🏘️ Medieval Centre & Red Light District [ Europe > The Netherlands > Amsterdam ]
🏘️ Amsterdam Noord                      [ Europe > The Netherlands > Amsterdam ]
🏞️ South Holland & Zeeland              [ Europe > The Netherlands ]            
🏘️ De Pijp                              [ Europe > The Netherlands > Amsterdam ]
🏞️ Friesland                            [ Europe > The Netherlands ]            
🏙️ Den Haag                             [ Europe > The Netherlands ]            
🏙️ Utrecht                              [ Europe > The Netherlands ]            
🏙️ Groningen                            [ Europe > The Netherlands ]            
🏙️ Maastricht                           [ Europe > The Netherlands ]            
🏞️ Frisian Islands                      [ Europe > The Netherlands ]            
🏙️ Nijmegen                             [ Europe > The Netherlands ]            
🏙️ Haarlem                              [ Europe > The Netherlands ]            
🏙️ Leiden                               [ Europe > The Netherlands ]            
🏙️ Delft                                [ Europe > The Netherlands ]            
🏙️ Eindhoven                            [ Europe > The Netherlands ]            ",
    ],
    level: "log",
  },
]
`;

snapshot[`lonely-planet [keywords...] --json 1`] = `
[
  {
    data: [
      '{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"}],"excerpt":"Tradition and innovation intertwine here: artistic masterpieces, centuries-old windmills, tulip fields and romantic candlelit cafés coexist with visionary architecture, cutting-edge design and phenomenal nightlife.","featuredImage":{"alt":"Traditional dutch windmills and houses near the canal in Zaanstad village, Zaanse Schans, Netherlands.","caption":"Traditional dutch windmills and houses near the canal in Zaanstad village, Zaanse Schans, Netherlands.","credit":"© Olena_Z/Getty Images/iStockphoto","height":3169,"name":"GettyImages-937057490.jpeg","title":"GettyImages-937057490.jpeg","url":"https://lp-cms-production.imgix.net/2023-03/GettyImages-937057490.jpeg","width":4765},"id":"24256","internalRanking":13.48548,"slug":"the-netherlands","title":"The Netherlands","type":"Country"}',
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"Things are different below the rivers: the provinces of Noord Brabant and Limburg have a character that's more Catholic, more relaxed, and in the case of southern Limburg, hillier than elsewhere in the Netherlands. Among the obvious southern influences are a preference for good beer and food, as embodied in the concept Bourgondisch: eating and drinking with a verve worthy of the epicurean inhabitants of France's Burgundy.","featuredImage":{"alt":"View over statue of Beeld van Minckelers on markt in Maastricht.\\\\n17th, architecture, attraction, belfort, belfry, benelux, blue, building, century, city, crowd, day, dusk, dutch, europe, european, evening, gathering, hall, historic, history, holland, huge, illumination, landmark, limburg, maastricht, market, netherlands, night, nobody, old, outside, people, pieter, restaurant, scenic, sightseeing, square, tourism, tourist, tower, town, travel, twilight, urban, view","caption":"View over statue of Beeld van Minckelers on markt in Maastricht.\\\\n17th, architecture, attraction, belfort, belfry, benelux, blue, building, century, city, crowd, day, dusk, dutch, europe, european, evening, gathering, hall, historic, history, holland, huge, illumination, landmark, limburg, maastricht, market, netherlands, night, nobody, old, outside, people, pieter, restaurant, scenic, sightseeing, square, tourism, tourist, tower, town, travel, twilight, urban, view","credit":"©pavel dudek/Shutterstock","height":1624,"name":"shutterstockRF303054407.jpg","title":"shutterstockRF303054407.jpg","url":"https://lp-cms-production.imgix.net/2023-09/shutterstockRF303054407.jpg","width":2525},"id":"24546","internalRanking":4.36551,"slug":"the-netherlands/southeastern-netherlands","title":"Southeastern Netherlands","type":"Region"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"Golden Age canals lined by tilting gabled buildings are the backdrop for Amsterdam's treasure-packed museums, vintage-filled shops and hyper-creative drinking, dining and design scenes.","featuredImage":{"alt":"Man riding bike through Amsterdam.","caption":"Man riding bike through Amsterdam.","credit":"© Raquel Arocena Torres/Getty Images","height":4000,"name":"man riding bike raquel arocena torres GettyImages-1272262535 rfc.jpg","title":"man riding bike raquel arocena torres GettyImages-1272262535 rfc.jpg","url":"https://lp-cms-production.imgix.net/2021-10/man%20riding%20bike%20raquel%20arocena%20torres%20GettyImages-1272262535%20rfc.jpg","width":6000},"id":"24251","internalRanking":11.37215,"slug":"the-netherlands/amsterdam","title":"Amsterdam","type":"City"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"Innovation is the mantra in the Netherlands' second city, and the locals embrace it with an enthusiasm that makes a visit here an exciting proposition.","featuredImage":{"alt":"Rotterdam's Delfshaven with his Historic Boats","caption":"Rotterdam's Delfshaven with his Historic Boats","credit":"Getty Images","height":2323,"name":"25b16feb8d0061ec61a188edd0577c26-delfshaven.jpg","title":"25b16feb8d0061ec61a188edd0577c26-delfshaven.jpg","url":"https://lp-cms-production.imgix.net/2019-06/25b16feb8d0061ec61a188edd0577c26-delfshaven.jpg","width":3499},"id":"188216","internalRanking":8.63551,"slug":"the-netherlands/rotterdam","title":"Rotterdam","type":"City"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"},{"slug":"the-netherlands/amsterdam","title":"Amsterdam","type":"City"}],"excerpt":"Amsterdam's oldest quarter is remarkably preserved, looking much as it did in its Golden Age heyday. It's the busiest part of town for visitors. While some come to see the Royal Palace and Oude Kerk, others make a beeline for the coffeeshops and Red Light District.","featuredImage":{"alt":"The Royal Palace in Amsterdam is situated in the west side of Dam Square in the centre of Amsterdam, Holland. Beautiful blue sky with cloudscape over the palace. The Netherlands.","caption":"The Royal Palace in Amsterdam is situated in the west side of Dam Square in the centre of Amsterdam, Holland. Beautiful blue sky with cloudscape over the palace. The Netherlands.","credit":"Getty Images/iStockphoto","height":4912,"name":"The Royal Palace, Amsterdam, Netherlands","title":"The Royal Palace, Amsterdam, Netherlands","url":"https://lp-cms-production.imgix.net/2022-02/iStock-476182609%20RFC.jpg","width":7360},"id":"24286","internalRanking":8.50106,"slug":"the-netherlands/amsterdam/medieval-centre-and-red-light-district","title":"Medieval Centre & Red Light District","type":"Neighborhood"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"},{"slug":"the-netherlands/amsterdam","title":"Amsterdam","type":"City"}],"excerpt":"Amsterdam Noord, a quick free ferry ride across the IJ River from central Amsterdam, is a previously neglected area that has been reinvented as the city's hippest neighbourhood. It encompasses ex-industrial areas, cutting-edge architecture, and hangars turned hipster hangouts with walls covered in street art, all minutes away from fields, horses and the odd windmill. Noord is perfect for exploring by bike, stopping to eat and drink in its alternative venues. It's a great area to escape the crowds of central Amsterdam, too.","featuredImage":{"alt":"AMSTERDAM, HOLLAND - AUGUST 01: EYE Film Institute and Overhoeks Tower in Amsterdam, Holland. August 01, 2012 Amsterdam, Holland. EYE Film Institute includes a cinematography museum.; Shutterstock ID 111061883; Your name (First / Last): Daniel Fahey; GL account no.: 65050; Netsuite department name: Online Editorial; Full Product or Project name including edition: Amsterdam-Noord neighbourhood","caption":"AMSTERDAM, HOLLAND - AUGUST 01: EYE Film Institute and Overhoeks Tower in Amsterdam, Holland. August 01, 2012 Amsterdam, Holland. EYE Film Institute includes a cinematography museum.; Shutterstock ID 111061883; Your name (First / Last): Daniel Fahey; GL account no.: 65050; Netsuite department name: Online Editorial; Full Product or Project name including edition: Amsterdam-Noord neighbourhood","credit":"Shutterstock / Igor Plotnikov","height":1702,"name":"0d3566c4e927e260ef944dacc04ed05a-amsterdam-noord.jpg","title":"0d3566c4e927e260ef944dacc04ed05a-amsterdam-noord.jpg","url":"https://lp-cms-production.imgix.net/2019-06/0d3566c4e927e260ef944dacc04ed05a-amsterdam-noord.jpg","width":3024},"id":"141996","internalRanking":6.10925,"slug":"the-netherlands/amsterdam/amsterdam-noord","title":"Amsterdam Noord","type":"Neighborhood"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"It's the contrasts that make South Holland and Zeeland such compelling destinations. One day, you're in the ongoing architectural experiment that is Rotterdam, marvelling at its ever-growing portfolio of cutting-edge buildings. The next day you might be in Delft, where the historic town centre is so well-preserved that you'll feel as if you've been transported back to Holland's 17th-century Golden Age. And the contrasts aren't confined to building stock: Den Haag has a sophisticated urban charm, enchanting visitors with its art, fine dining and cultural scene; Zeeland has a dramatic back-of-beyond feel, with a sparsely populated and windswept landscape; and Gouda plays to every Dutch stereotype, with wheels of its famous yellow cheese and traditionally garbed locals ever-present. The bottom line for travellers is that there's something for everyone here, whether it be windmills or wine bars, tulips or tasting menus.","featuredImage":{"alt":"The Westkapelse Zeedijk forms part of the dutch coastal flood defense system. This modern dike, with a layer of basalt stones covered with sand and grass, protects he whole island of Walcheren against coastal floods. Modern flood protection measures allow for recreational and economic activities and development of a natural environment. Picture is taken on a stormy day in summer when a thunderstorm is approaching.","caption":"The Westkapelse Zeedijk forms part of the dutch coastal flood defense system. This modern dike, with a layer of basalt stones covered with sand and grass, protects he whole island of Walcheren against coastal floods. Modern flood protection measures allow for recreational and economic activities and development of a natural environment. Picture is taken on a stormy day in summer when a thunderstorm is approaching.","credit":"Getty Images/iStockphoto","height":2083,"name":"Approaching thunderstorm and dutch coastal flood defense","title":"Approaching thunderstorm and dutch coastal flood defense","url":"https://lp-cms-production.imgix.net/2022-02/iStock-621839846%20RFC.jpg","width":5029},"id":"188126","internalRanking":5.05184,"slug":"the-netherlands/south-holland-zeeland","title":"South Holland & Zeeland","type":"Region"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"},{"slug":"the-netherlands/amsterdam","title":"Amsterdam","type":"City"}],"excerpt":"A hotbed of creativity, multicultural De Pijp is less than five minutes from Centraal Station by metro yet preserves its village-like atmosphere. The neighbourhood's centrepiece is Amsterdam's largest street market, the colourful Albert Cuypmarkt, and the fashion boutiques, vintage shops, experimental restaurants and free-spirited cafés (pubs) that surround it.","featuredImage":{"alt":"Sarphati Park streetscape in the De Pijp area.","caption":"Sarphati Park streetscape in the De Pijp area.","height":2606,"name":"cea3e9d621ccc174356d6d7c88a67745378a261eda501d7b7174e7ca785f104e.jpg","title":"cea3e9d621ccc174356d6d7c88a67745378a261eda501d7b7174e7ca785f104e.jpg","url":"https://lp-cms-production.imgix.net/2019-06/cea3e9d621ccc174356d6d7c88a67745378a261eda501d7b7174e7ca785f104e.jpg","width":4021},"id":"24266","internalRanking":4.58497,"slug":"the-netherlands/amsterdam/de-pijp","title":"De Pijp","type":"Neighborhood"}\`,
    ],
    level: "log",
  },
  {
    data: [
      '{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"This unspoilt northerly province is the spot to take your foot off the pedal and go slow, Friesian-style, amid raw coastal nature and a profusion of namesake black-and-white cows.","featuredImage":{"alt":"heather flowering at misty sunrise, Friesland, Netherlands; Shutterstock ID 331078466; Your name (First / Last): Daniel Fahey; GL account no.: 65050; Netsuite department name: Online Editorial; Full Product or Project name including edition: Friesland page","caption":"heather flowering at misty sunrise, Friesland, Netherlands; Shutterstock ID 331078466; Your name (First / Last): Daniel Fahey; GL account no.: 65050; Netsuite department name: Online Editorial; Full Product or Project name including edition: Friesland page","height":2000,"name":"1de75384ebec16691177cdaf71d3d8bd-friesland.jpg","title":"1de75384ebec16691177cdaf71d3d8bd-friesland.jpg","url":"https://lp-cms-production.imgix.net/2019-06/1de75384ebec16691177cdaf71d3d8bd-friesland.jpg","width":3000},"id":"24466","internalRanking":3.80444,"slug":"the-netherlands/friesland","title":"Friesland","type":"Region"}',
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"There's a lot more to Den Haag than immediately meets the eye. The popular perception of the Netherlands' third-largest city is of a stately, regal place populated with bureaucrats and businesspeople. While this is true to some extent, there is so much more: the city's cultural scene – anchored by the presence of the world-renowned Mauritshuis museum and Nederlands Dans Theater (and about to be boosted by the opening of the Spuiplein cultural precinct) – is one of the most exciting in the country; its culinary scene is replete with contemporary restaurants expertly experimenting with modern European cuisine; and its entertainment scene has moved far past the embassy cocktail parties that once predominated, making the party precinct of Grote Markt and the much-loved Paard live-music venue essential stops for every visitor. Easy to explore on foot or by tram, this is a city that amply rewards those who stay for a few days.","featuredImage":{"alt":"Vredespaleis The Hague by night","caption":"Vredespaleis The Hague by night","height":3840,"name":"76842393.jpg","title":"76842393.jpg","url":"https://lp-cms-production.imgix.net/2019-06/76842393.jpg","width":5760},"id":"188776","internalRanking":3.1043,"slug":"the-netherlands/the-randstad/den-haag","title":"Den Haag","type":"City"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"It's hard not to fall in love with Utrecht, one of the Netherlands' oldest urban centres and for centuries its religious heart. Now a vibrant university city, the compact medieval core radiates out from the iconic 15th-century Domtoren, ringed by a loop of very pretty tree-lined canals. Their central sections have distinctive double-level sides inset with what were once medieval warehouses – many now forming fascinating venues to eat, drink, dance or sleep, with terrace-walkways that extend right to the waterside. The brilliant cafe culture goes well beyond the canals – Utrecht's vibrant concert schedules are a big draw and, if the budget stretches, it's worth luxuriating in one of numerous historic mansions that have been turned into indulgent boutique hotels.","featuredImage":{"height":2952,"name":"57386144.jpg","title":"57386144.jpg","url":"https://lp-cms-production.imgix.net/2019-06/57386144.jpg","width":4428},"id":"150766","internalRanking":2.39132,"slug":"the-netherlands/the-randstad/utrecht-city","title":"Utrecht","type":"City"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"Looking at a map of the Netherlands, Groningen seems a long way from anywhere – but this vibrant, youthful city is among the country's most progressive urban metropolises, with a 40,000-strong student population injecting bags of creative zest into its buzzing cafe culture, hedonistic nightlife and rampant cultural scene.","featuredImage":{"alt":"Ships in Groningen","caption":"Ships in Groningen","credit":"Getty Images/iStockphoto","height":3348,"name":"GettyImages-460387209_super.jpg","title":"GettyImages-460387209_super.jpg","url":"https://lp-cms-production.imgix.net/2019-06/GettyImages-460387209_super.jpg","width":4968},"id":"159436","internalRanking":2.21976,"slug":"the-netherlands/the-north-and-east/groningen-city","title":"Groningen","type":"City"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"Lively and energetic, Maastricht has Roman history, a maze of tunnel-caves and historical buildings aplenty, plus a Burgundian sophistication to its dining, a bacchanalian delight to its drinking culture, and a student-friendly street-life out of all proportion to its size. The people are irreverent, shrugging off the shackles of Dutch restraint while speaking a dialect ('Mestreechs') that baffles Hollanders. No visit to the Netherlands is complete without a visit.","featuredImage":{"alt":"The Netherlands Maastricht at dusk","caption":"The Netherlands Maastricht at dusk","credit":"Getty Images/iStockphoto","height":4000,"name":"GettyImages-514855827_super.jpg","title":"GettyImages-514855827_super.jpg","url":"https://lp-cms-production.imgix.net/2019-06/GettyImages-514855827_super.jpg","width":6000},"id":"159346","internalRanking":2.10812,"slug":"the-netherlands/the-southeast/maastricht","title":"Maastricht","type":"City"}\`,
    ],
    level: "log",
  },
  {
    data: [
      '{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"The crescent of islands over Friesland – Vlieland, Terschelling, Ameland and Schiermonnikoog – form a unique natural entity and a distinct Dutch region. A natural barrier between the Frisian coast and the North Sea, they hem in the mudflats of the Waddenzee, a Unesco World Heritage site since 2009 and a hot spot for wadlopen (mudflat walking) with professional guides. Villages, polders and salt marshes fringe the islands along their Waddenzee shores, while swaths of beach and dunes lace the seaward side. Inland, expect ample forest and heath crisscrossed with hiking and cycling trails.","featuredImage":{"alt":"view of the Ameland Lighthouse, known as Bornrif, is a lighthouse on the Dutch island Ameland, one of the Frisian Islands, on the edge of the North Sea, The Netherlands","caption":"view of the Ameland Lighthouse, known as Bornrif, is a lighthouse on the Dutch island Ameland, one of the Frisian Islands, on the edge of the North Sea, The Netherlands","credit":"©Melanie Lemahieu/Shutterstock","height":2362,"name":"shutterstockRF_331097798.jpg","title":"shutterstockRF_331097798.jpg","url":"https://lp-cms-production.imgix.net/2022-02/shutterstockRF_331097798.jpg","width":3543},"id":"24476","internalRanking":1.91003,"slug":"the-netherlands/frisian-islands","title":"Frisian Islands","type":"Region"}',
    ],
    level: "log",
  },
  {
    data: [
      '{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"From Roman camp to Catholic university town of 13,000 students, Nijmegen buzzes with energy and is vibrant place to spend a couple of days. This is the oldest town in the Netherlands, with a compact medieval centre yet all the urban-cool of a big city. Historic cafes, specialist craft-coffee shops and a decent smattering of museums and monuments lace its streets that stagger gently from the 16th-century Lower Town by the banks of the Waal River to the older Upper Town. In summer, Dutch party lovers pour into town for its annual four-day hiking fest and Zomerfeesten music festival.","featuredImage":{"alt":"Central town square in Nijmegan","caption":"Central town square in Nijmegan","credit":"Getty Images/Lonely Planet Images","height":3267,"name":"GettyImages-475978935_high.jpg","title":"GettyImages-475978935_high.jpg","url":"https://lp-cms-production.imgix.net/2019-06/GettyImages-475978935_high.jpg","width":4900},"id":"24386","internalRanking":1.84839,"slug":"the-netherlands/nijmegen","title":"Nijmegen","type":"City"}',
    ],
    level: "log",
  },
  {
    data: [
      '{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"This classic Dutch city of cobbled streets, historic buildings, grand churches, even grander museums, cosy bars, fine cafes and canals is just a 15-minute train ride from Amsterdam. To its west are the coastal dunes of the Zuid-Kennemerland National Park and the popular beaches of Zandvoort and Bloemendaal aan Zee.","featuredImage":{"alt":"Postcard from Haarlem","caption":"Postcard from Haarlem","credit":"Getty Images/Moment Open","height":3776,"name":"GettyImages-539970901_super.jpg","title":"GettyImages-539970901_super.jpg","url":"https://lp-cms-production.imgix.net/2019-06/GettyImages-539970901_super.jpg","width":5664},"id":"188646","internalRanking":1.7525,"slug":"the-netherlands/the-randstad/haarlem","title":"Haarlem","type":"City"}',
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"Vibrant Leiden is one of the Netherlands' great cities. Woven with canals lined by beautiful 17th-century buildings, its cache of museums, all within walking distance of each other, are a major draw, as is wandering the picturesque canals and soaking up the nightlife, fuelled by a 27,000-strong student population.","featuredImage":{"alt":"Netherlands, Leiden, view across canal with Leiden university's Academy Building on the right","caption":"Netherlands, Leiden, view across canal with Leiden university's Academy Building on the right","credit":"Getty Images/Dorling Kindersley","height":3346,"name":"GettyImages-175133284_full.jpg","title":"GettyImages-175133284_full.jpg","url":"https://lp-cms-production.imgix.net/2019-06/GettyImages-175133284_full.jpg","width":5019},"id":"188146","internalRanking":1.2904,"slug":"the-netherlands/the-randstad/leiden","title":"Leiden","type":"City"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"An amalgam of austere medieval magnificence and Golden Age glory, Delft's exquisite town centre is a hugely popular Dutch day-trip destination, awash with visitors strolling its narrow, canal-lined streets and central Markt. The centre is time-capsule-like, having changed little since Golden Age artist Johannes Vermeer, who was born in Delft and lived his whole life here, painted his famous View of Delft in 1660–61.","featuredImage":{"alt":"A canal in the historic centre of Delft","caption":"A canal in the historic centre of Delft","credit":"Getty Images","height":3744,"name":"GettyImages-500741461_full.jpg","title":"GettyImages-500741461_full.jpg","url":"https://lp-cms-production.imgix.net/2019-06/GettyImages-500741461_full.jpg","width":5616},"id":"188156","internalRanking":1.22963,"slug":"the-netherlands/the-randstad/delft","title":"Delft","type":"City"}\`,
    ],
    level: "log",
  },
  {
    data: [
      \`{"breadcrumb":[{"slug":"europe","title":"Europe","type":"Continent"},{"slug":"the-netherlands","title":"The Netherlands","type":"Country"}],"excerpt":"If you find yourself in the Netherlands' fifth-biggest city it's probably for an international budget flight or to watch a football match at PSV's Philips Stadium. But Eindhoven is also a vibrant centre for creative design and post-industrial reinvention. OK, so visually the townscape has all the charm of Minsk on a bad hair day, but before rushing on, do at least consider visiting the fascinating Philips Museum, which tells the story of the global electronics company that essentially created modern Eindhoven. There's also a highly thought-provoking modern-art gallery and a UFO-shaped quirk of 1960s architecture.","featuredImage":{"alt":"Waterfront buildings at Eindhoven city in the Netherlands.","caption":"Waterfront buildings at Eindhoven city in the Netherlands.","credit":"©Juan Monte/EyeEm/Getty Images","height":3994,"name":"Panoramic View Of City Against Sky","title":"Panoramic View Of City Against Sky","url":"https://lp-cms-production.imgix.net/2022-02/GettyRF_742338977.jpg","width":6000},"id":"24596","internalRanking":0.7993,"slug":"the-netherlands/eindhoven","title":"Eindhoven","type":"City"}\`,
    ],
    level: "log",
  },
  {
    data: [
      "",
    ],
    level: "log",
  },
]
`;

snapshot[`lonely-planet --destinations 1`] = `
[
  {
    data: [
      "🏙️ Amsterdam                            [ Europe > The Netherlands ]            
🏘️ Amsterdam Noord                      [ Europe > The Netherlands > Amsterdam ]
🏘️ Medieval Centre & Red Light District [ Europe > The Netherlands > Amsterdam ]
🏘️ De Pijp                              [ Europe > The Netherlands > Amsterdam ]",
    ],
    level: "log",
  },
]
`;

snapshot[`lonely-planet --attractions 1`] = `
[
  {
    data: [
      "🎡 Frans Hals Museum – Hof         [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Grote Kerk van St Bavo          [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Zuid-Kennemerland National Park [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Frans Hals Museum – Hal         [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Teylers Museum                  [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Haarlemmerhout                  [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Corrie ten Boom House           [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Erebegraafplaats Bloemendaal    [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 De Zandwaaier Visitor Centre    [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Stadhuis                        [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Bakenesserkerk                  [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Proveniershuis                  [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Laurens Coster Statue           [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Nieuwe Kerk                     [ Europe > The Netherlands > Haarlem > Attractions ]                       
🎡 Haarlemmerpoort                 [ Europe > The Netherlands > Amsterdam > Jordaan & the West > Attractions ]",
    ],
    level: "log",
  },
]
`;

snapshot[`lonely-planet --stories 1`] = `
[
  {
    data: [
      "📰 How to spend a perfect day in Utrecht",
    ],
    level: "log",
  },
]
`;
